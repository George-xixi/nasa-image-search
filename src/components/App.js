import React from "react";
import { useState } from "react";
import Search from "./Search";
import SearchResults from "./SearchResults";
import "../styles/app.css";

const App = () => {
  const [searchResults, setSearchResults] = useState([]);
  const [loading, setLoading] = useState(true);
    
  return (
    <div className="app">
      <img className="nasa-logo"
        src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg"
        alt="red nasa logo"
      />
      <Search setSearchResults={setSearchResults} setLoading={setLoading} />
      {!loading && <SearchResults results={searchResults} /> }
    </div>
  );
};

export default App;
