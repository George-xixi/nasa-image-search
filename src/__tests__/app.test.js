import { render, screen } from "@testing-library/react";
import App from "../components/App";

describe("App", () => {
  it("renders App correctly", () => {
    const { asFragment } = render(<App />);
    const nasaImg = screen.getByAltText("The word NASA in red");

    expect(asFragment()).toMatchSnapshot();
    expect(nasaImg).toHaveAttribute("src");
  });
});
