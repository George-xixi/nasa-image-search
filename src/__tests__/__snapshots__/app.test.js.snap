// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App correctly 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <img
      alt="red nasa logo"
      class="nasa-logo"
      src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg"
    />
    <form
      class="search-form"
    >
      <input
        class="search-input"
        data-testid="search-input"
        type="text"
      />
      <button
        class="search-button"
        data-testid="search-button"
        type="submit"
      >
        ?
      </button>
    </form>
    <p>
      No results
    </p>
  </div>
</DocumentFragment>
`;
